packages/core/src/behaviors.ts

```diff
@@ -0,0 +1,111 @@
+import { ActorContext, ActorRef, Behavior, EventObject, Observer } from '.';
+import { doneInvoke } from './actions';
+import { toActorRef } from './Actor';
+import { toObserver } from './utils';
+
+/**
+ * Returns an actor behavior from a reducer and its initial state.
+ *
+ * @param transition The pure reducer that returns the next state given the current state and event.
+ * @param initialState The initial state of the reducer.
+ * @returns An actor behavior
+ */
+export function fromReducer<TState, TEvent extends EventObject>(
+  transition: (
+    state: TState,
+    event: TEvent,
+    actorContext: ActorContext<TEvent, TState>
+  ) => TState,
+  initialState: TState
+): Behavior<TEvent, TState> {
+  return {
+    transition,
+    initialState
+  };
+}
+
+export function fromPromise<T>(
+  promiseFn: () => Promise<T>
+): Behavior<{ type: 'resolve'; data: T }, T | undefined> {
+  return {
+    transition: (state, event, { parent, id }) => {
+      switch (event.type) {
+        case 'resolve':
+          parent?.send(doneInvoke(id, event.data));
+          return event.data;
+        default:
+          return state;
+      }
+    },
+    initialState: undefined,
+    start: ({ self }) => {
+      promiseFn().then((data) => {
+        self.send({ type: 'resolve', data });
+      });
```

Comment:
We should also handle errors here and add handling of that event
```suggestion
      promiseFn().then(
        (data) => {
          self.send({ type: 'resolve', data });
        },
        (error) => {
          self.send({ type: 'reject', error });
        }
      );
```